lib/python/script/utils.py

```diff
@@ -467,3 +471,104 @@ def clock():
     if sys.version_info > (3,2):
         return time.perf_counter()
     return time.clock()
+
+
+def legalize_vector_name(name):
+    if not name:
+        raise ValueError("name cannot be empty")
+    if re.match("[^A-Za-z0-9]", name[0]):
+        name[0] = "_"
+    re.sub("[^A-Za-z0-9_]", "_", name)
+    keywords = ["and", "or", "not"]
```

Comment:
There are a few more other keywords that can cause problems, esp. if the scope is also temporary names for usage in GRASS SQL modules (without quoted identifiers). Most standard SQL keywords are probably listed here: https://www.w3schools.com/sql/sql_ref_keywords.asp
I don`t think it is possible to catch all of them (esp. with differnt backends that add to them), so having e.g. an underscore as a mandatory part of the name would help to avoid such issues and does no harm to temp names.
Alternatively,
```
import keyword
keyword.iskeyword(name)
```
and 
```
import builtins
name in dir(builtins)
```
could be used for Python keywords (https://stackoverflow.com/questions/22864221/is-the-list-of-python-reserved-words-and-builtins-available-in-a-library)

Reply:
I'm not sure what to do. The current list comes from [`Vect_legal_filename()`](https://grass.osgeo.org/programming7/legal__vname_8c.html#afb15448628c539456b751848c460fcd0), so we would need some solution for SQL keywords in C as well to keep this in sync. I was surprised how short the list is there. Perhaps a new issue?